Mathlib/CategoryTheory/MorphismProperty/Lifting.lean

```diff
@@ -0,0 +1,159 @@
+/-
+Copyright (c) 2024 Jack McKoen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jack McKoen
+-/
+import Mathlib.CategoryTheory.MorphismProperty.Basic
+import Mathlib.CategoryTheory.MorphismProperty.Limits
+
+/-!
+# Left and right lifting properties
+
+Given a morphism property `T`, we define the left and right lifting property with respect to `T`.
+
+We show that the left lifting property is stable under retracts, cobase change, and composition,
+with dual statements for the right lifting property.
+
+-/
+
+universe v u
+
+namespace CategoryTheory
+
+variable {C : Type u} [Category.{v} C]
+
+/-- `f : X ⟶ Y` is a retract of `g : Z ⟶ W` if there are morphisms `i : f ⟶ g`
+and `r : g ⟶ f` in the arrow category of `C` such that `i ≫ r = 𝟙 f`. -/
+class IsRetract {X Y Z W : C} (f : X ⟶ Y) (g : Z ⟶ W) where
```

Comment:
As this contains data, this should not be named `IsRetract`. Also, this notion should be defined in any category, not just the `Arrow` category.

Reply:
(Maybe use an abbrev `RetractArrow {X Y Z W : C} (f : X ⟶ Y) (g : Z ⟶ W) : Prop` for the arrow variant.

Reply:
Basic notions about retracts could to in a file like `CategoryTheory.Retract`.